test/integration/test-amp-inabox.js

```diff
@@ -64,33 +164,177 @@ describe('inabox', function() {
+    });
+
+    it.configure().skipSafari().run('should layout amp-img, amp-pixel, ' +
+        'amp-analytics within safe frame', () => {
+      writeSafeFrame(env.win.document, iframe, adContent);
+      return testAmpComponentsBTF(env.win);
+    });
+  });
+});
+
+describe('inabox with a complex image ad', function() {
+  // The image ad as seen in examples/inabox.gpt.html,
+  // with visibility pings being placeholders that's substituted with calls to
+  // the request bank.
+  const adBody = __html__['test/fixtures/amp-cupcake-ad.html'] // eslint-disable-line no-undef
+      .replace(/__VIEW_URL__/g, RequestBank.getUrl('view')) // get all instances
+      .replace('__VISIBLE_URL__', RequestBank.getUrl('visible'))
+      .replace('__ACTIVE_VIEW_URL__', RequestBank.getUrl('activeview'));
+
+  function testVisibilityPings(visibleDelay, activeViewDelay) {
+    let viewTime = 0;
+    let visibleTime = 0;
+    let activeViewTime = 0;
+    const viewPromise = RequestBank.withdraw('view')
+        .then(() => viewTime = Date.now());
+    const visiblePromise = RequestBank.withdraw('visible')
+        .then(() => visibleTime = Date.now());
+    const activeViewPromise = RequestBank.withdraw('activeview')
+        .then(() => activeViewTime = Date.now());
+    return Promise.all([viewPromise, visiblePromise, activeViewPromise])
+        .then(() => {
+          // Add a 200ms "buffer" to account for possible browser jankiness
+          expect(visibleTime - viewTime).to.be.above(visibleDelay - 200);
+          expect(activeViewTime - viewTime).to.be.above(activeViewDelay - 200);
+        });
+  }
+
+  describes.integration('AMP Inabox Rendering', {
+    amp: false,
+    body: `
+        <script src="/examples/inabox-tag-integration.js"></script>
+        `,
+  }, env => {
+    let iframe;
+    let doc;
+    beforeEach(() => {
+      doc = env.win.document;
+      iframe = document.createElement('iframe');
+      // we add the iframe here because it's dynamically created, so the
+      // bootstrap script would have missed it.
+      Array.prototype.push.apply(env.win.top.ampInaboxIframes, [iframe]);
+    });
+
+    it('should properly render ad in a friendly iframe with viewability pings',
+        () => {
+          writeFriendlyFrame(doc, iframe, adBody);
+          return testVisibilityPings(0, 1000);
+        });
+
+    it('should properly render ad in a safe frame with viewability pings',
+        () => {
+          writeSafeFrame(doc, iframe, adBody);
+          return testVisibilityPings(0, 1000);
+        });
+
+    afterEach(() => {
+      doc.body.removeChild(iframe);
+    });
+  });
+
+  // Testing that analytics components use IntersectionObserver properly.
+  describes.realWin('AMP Inabox Rendering - No Host Script', {
+    amp: false,
+  }, env => {
+    let iframe;
+    let doc;
+    beforeEach(() => {
+      doc = env.win.document;
+      iframe = document.createElement('iframe');
+    });
+
+    it.configure().skipSafari().run(
```

Comment:
i'm not 100% sure if that will work, but could you try to move `configure.skipSafari()` to `describes`, like `describes.configure.skipSafari().realwin`.

Reply:
I tried and it doesn't work; only describe and it has configure() added to their prototypes in `test/_init.test.js`. We'll have to patch `testing/describes.js` to add that functionality.

Reply:
too bad :-(